### PR TITLE
clean todo on 000 it 

### PR DESCRIPTION
**Pull request by gfournieriExec**

### CODE REVIEW

**Review by jeremyjams — APPROVED** [review]
Thank you :)

File: test/000_fullchain.test.ts [hunk 1]
```diff
@@ -126,36 +127,34 @@ describe('Integration tests', function () {
...
-                worker1,
-            );
-            await iexecPoco
-                .connect(worker1)
-                .reveal(taskId, callbackResultDigest)
-                .then((tx) => tx.wait());
-            await iexecPoco
+            const workerStakePerTask = await iexecPoco
+                .viewDeal(dealId)
+                .then((deal) => deal.workerStake.toNumber());
```
Comment (zguesmi) [inline test/000_fullchain.test.ts:139]: This can go outside of the `for` loop.

File: test/000_fullchain.test.ts [hunk 2]
```diff
@@ -126,36 +127,34 @@ describe('Integration tests', function () {
...
+                await iexecPoco
+                    .connect(worker)
+                    .reveal(taskId, callbackResultDigest)
+                    .then((tx) => tx.wait());
+            }
+            const finalizeTx = await iexecPoco
                 .connect(scheduler)
-                .finalize(taskId, results, resultsCallback)
-                .then((tx) => tx.wait());
+                .finalize(taskId, results, resultsCallback);
```
Comment (zguesmi) [inline test/000_fullchain.test.ts:157]:
```suggestion
                .finalize(taskId, results, resultsCallback);
            await finalizeTx.wait();
```

File: test/000_fullchain.test.ts [hunk 3]
```diff
@@ -174,44 +173,380 @@ describe('Integration tests', function () {
             //   - Worker:
```
Comment (zguesmi) [inline test/000_fullchain.test.ts:173]: What do you think of removing this comment block ? Balances check seems clearer now.

File: test/000_fullchain.test.ts [hunk 4]
```diff
@@ -174,44 +173,380 @@ describe('Integration tests', function () {
...
+            }
+            // Calculate expected frozen changes
+            const expectedFrozenChanges = [
+                0, // Proxy
+                -taskPrice * completedTasks, // Sponsor
+                0, // Requester
+                -schedulerStakePerTask * completedTasks, // Scheduler
+                0, // AppProvider
+                0, // DatasetProvider
+            ];
```
Comment (zguesmi) [inline test/000_fullchain.test.ts:208]:
```suggestion
                0, // DatasetProvider
                ...workers.map(() => 0) // Add 0 for each worker
            ];
```
And remove the for loop below.

File: test/000_fullchain.test.ts [hunk 5]
```diff
@@ -174,44 +173,380 @@ describe('Integration tests', function () {
...
+                    datasetPrice, // DatasetProvider
                 ],
+            );
+            for (const worker of workers) {
+                expect(finalizeTx).to.changeTokenBalances(
+                    iexecPoco,
+                    [worker],
+                    [workerStakePerTask + workerRewardPerTask / workers.length],
+                );
+            }
```
Comment (zguesmi) [inline test/000_fullchain.test.ts:199]: We can integrate this into the check above:
```ts
            const expectedWorkerBalanceChange = workerStakePerTask + workerRewardPerTask / workers.length;
            expect(finalizeTx).to.changeTokenBalances(
                iexecPoco,
                [proxyAddress, [...] , datasetProvider, ...workers],
                [
                    expectedProxyBalanceChange, // Proxy
                    [...]
                    datasetPrice, // DatasetProvider
                    ...workers.map(() => expectedWorkerBalanceChange) // Workers
                ],
            );
```

File: test/000_fullchain.test.ts [hunk 6]
```diff
@@ -174,44 +173,380 @@ describe('Integration tests', function () {
...
+                -taskPrice * completedTasks, // Sponsor
+                0, // Requester
+                -schedulerStakePerTask * completedTasks, // Scheduler
+                0, // AppProvider
+                0, // DatasetProvider
+            ];
+            for (let i = 0; i < workers.length; i++) {
+                expectedFrozenChanges.push(0);
+            }
+            await changesInFrozen({
```
Comment (zguesmi) [inline test/000_fullchain.test.ts:212]:
```suggestion
            await checkFrozenChanges({
```

File: test/000_fullchain.test.ts [hunk 7]
```diff
@@ -416,21 +749,6 @@ describe('Integration tests', function () {
...
-            account.balance,
-            message,
-        );
-        expect(await iexecPoco.frozenOf(account.signer.address)).to.equal(account.frozen, message);
-    }
-}
-
 async function changesInFrozen(args: {
     accountsInitialFrozens: { address: string; frozen: number }[];
     frozenChanges: number[];
```
Comment (zguesmi) [inline test/000_fullchain.test.ts:754]: Why `args`?
```suggestion
async function changesInFrozen(
    accountsInitialFrozens: { address: string; frozen: number }[],
    frozenChanges: number[],
```

File: test/000_fullchain.test.ts (hunk 5 above)
Comment (gfournieriExec) [reply test/000_fullchain.test.ts]: [check all in one changeTokenBalances with workers](https://github.com/iExecBlockchainComputing/PoCo/pull/163/commits/1cfd890aaa09ae6e75fc9418fbf1424ab02f360d)

File: test/000_fullchain.test.ts (hunk 4 above)
Comment (gfournieriExec) [reply test/000_fullchain.test.ts]: [push workers frozen balance in the array directly](https://github.com/iExecBlockchainComputing/PoCo/pull/163/commits/387e4a4ccf72da1c1be89d9bad674f35ce8bd139)

File: test/000_fullchain.test.ts (hunk 3 above)
Comment (gfournieriExec) [reply test/000_fullchain.test.ts]: [remove comment block](https://github.com/iExecBlockchainComputing/PoCo/pull/163/commits/7aa37cbcf474ea39b7ff2b33bdee8064736fef5c)

File: test/000_fullchain.test.ts (hunk 1 above)
Comment (gfournieriExec) [reply test/000_fullchain.test.ts]: [set workerStakePerTask outside of the loop](https://github.com/iExecBlockchainComputing/PoCo/pull/163/commits/aeeba74c359210e08fca73f8efe80cdf2d148475)

**Pull request merged**